operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -189,6 +193,13 @@ private String getLockName(String namespace, String name) {
             T cr = resourceOperator.get(namespace, name);
 
             if (cr != null) {
+                if (!Util.matchesSelector(selector(), cr))  {
+                    // The the labels matching the selector are removed, the DELETED event is triggered. We have to
+                    // filter these situations out and ignore the reconciliation.
+                    log.debug("{}: {} {} in namespace {} does not match label selector {} and will be ignored", reconciliation, kind(), name, namespace, selector().get().getMatchLabels());
```

Comment:
can you elaborate a little bit on the comment? It's not clear to me. This is the case where the CR exists but doesn't match the label selectors so we are ignoring it. How the comment is related to this?

Reply:
The comment is there because this is tricky and possibly unexpected. The watch is configured to watch for resources with some labels. So one would expect to get only events when the resource has these labels. But that is not the case. If you remove the label from the CR, it triggers a DELETE event for the watch despite the resource not being technically deleted and the CR labels not matching the watch filter anymore. So we need to filter these situations out and that is why this additional check had to be added there. 

And because I think it is not obvious why we check labels again in a watch which already has them in the selector I added the comment there

Reply:
I updated the comment to make it a bit more clear.

Reply:
This triggered DELETE is something "bad" in K8S API server or fabric8 watcher? Doesn't sound normal to trigger a DELETE because labels are removed :-o

Reply:
That would not be Fabric8 but Kube API server if something. But I think this is a correct and desired behaviour. Essentially, by deleting the labels, you delete the resource from the watch. So I think it makes sense and there will be many cases where this would be actually important (even in Strimzi, I imagine in User or Topic Operators or in the Connector watch this is exactly what we want). But in our use-case with the handover it does not work so we need to identify it and ignore it.

Reply:
Tbh don't know if it's really correct. You are actually removing it from the watcher so "not watching it anymore" but the resource is still in etcd and a DELETE event should be related to what happens to the resource in etcd I think.
Anyway, there is nothing we can do, we can just deal with it.

Reply:
Ok, great. In that case your comments should be incorporated.